assets/src/dashboard/app/views/myStories/karma/myStories.karma.js

```diff
@@ -436,6 +504,182 @@ describe('Grid view', () => {
       expect(renderedStoriesById).toEqual(storiesOrderById);
     });
   });
+
+  describe('Dashboard keyboard navigation and focus logic', () => {
+    it('should navigate the grid via keyboard', async () => {
+      const storyCards = fixture.screen.getAllByTestId(/card-action-container/);
+      await focusOnGridByKeyboard();
+
+      for (let i = 0; i < storyCards.length; i++) {
+        const expectedStory = storyCards[i];
+        // eslint-disable-next-line no-await-in-loop
+        await fixture.events.keyboard.press('right');
+        expect(expectedStory).toEqual(document.activeElement);
+      }
+    });
+
+    it('should focus on context menu items via keyboard', async () => {
+      const storyCards = fixture.screen.getAllByTestId(/card-action-container/);
+      const selectedStory = storyCards[0];
+      await focusOnGridByKeyboard();
+      await fixture.events.keyboard.press('right');
+
+      // we should have focused the first story in the grid
+      expect(selectedStory).toEqual(document.activeElement);
+
+      let limit = 0;
+      const [contextMenuKabab] = fixture.screen.getAllByLabelText(
+        'More Options'
+      );
+
+      // tab to the button which opens the context menu
+      while (!contextMenuKabab.contains(document.activeElement) && limit < 4) {
+        // eslint-disable-next-line no-await-in-loop
+        await fixture.events.keyboard.press('tab');
+        limit++;
+      }
+
+      expect(contextMenuKabab).toEqual(contextMenuKabab);
+
+      await fixture.events.keyboard.press('Enter');
+
+      // now the focused item should be the first context menu item
+      const [contextMenuList] = fixture.screen.getAllByTestId(
+        /context-menu-list/
+      );
+      // it is focused on the link within the list
+      const [firstContextMenuItem, secondContextMenuItem] = within(
+        contextMenuList
+      ).getAllByText((content, element) => {
+        return element.tagName.toLowerCase() === 'a';
+      });
```

Comment:
💭 could we do a `getAllByRole('a')` here instead? Since we're selecting by tag I think `ByRole` would make more sense

Reply:
Well, `'a'` isn't a role, the `a` tag is not a role, which is why I used the tag name. Since I don't think it's accurate that it's not a `link` or a `button` (depending on what each menu item does), so I applied the `role="menuitem"` to the `MenuItem` and role `role="menu"` to the menu list to be axe compliant. This also meant removing aria-expanded to the MenuList because aria-expanded is not supported on the `menu` role.